packages/camera/camera_android/lib/src/android_camera.dart

```diff
@@ -540,9 +546,14 @@ class AndroidCamera extends CameraPlatform {
         return 'medium';
       case ResolutionPreset.low:
         return 'low';
-      default:
-        throw ArgumentError('Unknown ResolutionPreset value');
     }
+    // The enum comes from a different package, which could get a new value at
+    // any time, so provide a fallback that ensures this won't break when used
+    // with a version that contains new values. This is deliberately outside
+    // the switch rather than a `default` so that the linter will flag the
+    // switch as needing an update.
+    // ignore: dead_code
+    return 'max';
```

Comment:
Do you think we should just keep the old behavior of throwing?  It seems like in most of these cases you can keep the same behavior by removing the exception outside of the default branch which would keep the same logic.

Reply:
We could keep the same logic, but I changed it because throwing seems very extreme to me when there's a reasonable fallback.

When these switches were first written, everything was one package and this case meant that we'd forgotten to implement something, so it was a bug. Now it means that a client is using an implementation version that doesn't support that option yet, so a fallback seems better to me.